src/Tasks/GetCompatiblePlatform.cs

```diff
@@ -0,0 +1,153 @@
+                if (childPlatformLookupTable != null &&
+                        childPlatformLookupTable.ContainsKey(CurrentProjectPlatform) &&
+                        childPlatforms.Contains(childPlatformLookupTable[CurrentProjectPlatform]))
+                {
+                    buildChildProjectAs = childPlatformLookupTable[CurrentProjectPlatform];
+                    Log.LogMessage($"Found '{CurrentProjectPlatform}={buildChildProjectAs}' in the referenced project's translation table.");
+                }
+                // Current project's translation table follows
+                else if (translationTable != null &&
+                        translationTable.ContainsKey(CurrentProjectPlatform) &&
+                        childPlatforms.Contains(translationTable[CurrentProjectPlatform]))
+                {
+                    buildChildProjectAs = translationTable[CurrentProjectPlatform];
+                    Log.LogMessage($"Found '{CurrentProjectPlatform}={buildChildProjectAs}' in the current project's translation table.");
+                }
+                // Prefer matching platforms
+                else if (childPlatforms.Contains(CurrentProjectPlatform))
+                {
+                    buildChildProjectAs = CurrentProjectPlatform;
+                    Log.LogMessage($"Child and parent have the same platform.");
+                }
+                // AnyCPU if possible
+                else if (childPlatforms.Contains("AnyCPU"))
+                {
+                    buildChildProjectAs = "AnyCPU";
+                    Log.LogMessage($"Defaulting to AnyCPU.");
+                }
+                else
+                {
+                    // Keep NearestPlatform empty, log a warning. Common.CurrentVersion.targets will undefine 
+                    // Platform/PlatformTarget when this is the case.
+                    Log.LogWarningWithCodeFromResources("GetCompatiblePlatform.NoCompatiblePlatformFound", AssignedProjectsWithPlatform[i].ItemSpec);
+                }
+
+                AssignedProjectsWithPlatform[i].SetMetadata("NearestPlatform", buildChildProjectAs);
+                Log.LogMessage($"Project '{AssignedProjectsWithPlatform[i].ItemSpec}' will build with Platform: '{buildChildProjectAs}'");
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private Dictionary<string, string> ExtractLookupTable(string stringTable)
+        {
+            if (string.IsNullOrEmpty(stringTable))
+            {
+                return null;
+            }
+
+            Dictionary<string, string> table = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (string s in stringTable.Split(MSBuildConstants.SemicolonChar, StringSplitOptions.RemoveEmptyEntries))
+            {
+                string[] keyVal = s.Split(MSBuildConstants.EqualsChar, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Only unnecessary errors in strange translation tables like `A=Z;;B=G` that are "technically" correct.

Reply:
For the semicolon split I'm totally on board. For the equals I'm still pretty skeptical.

Reply:
Sounds reasonable. Will allow empty entries on splitting the equals.

Reply:
Allowing empty entries complicates the invalid check and doesn't get us much benefit. Disallowing empty entries guarantees `a=b` at minimum, and keeps the invalid check simpler. Otherwise its an extra two string null/empty checks, which would have been handled by string.Split

Reply:
But it also silently handles super confusing cases like `=x86========anycpu==`, which I think is bad.

Reply:
> But it also silently handles super confusing cases like =x86========anycpu==, which I think is bad.

Blarg, blarg I say! (Translation: you're right, will fix)